### PR TITLE
ethernet: smsc91x: rework the device node hierarchy

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Hi, various cleanups on the smsc91x driver, mainly reworking the mdio/phy hierarchy so that we can match up the init sequence but also various cleanups I bumped into.

Note that I had to hack some `enabled = "okay"` to test this, it does not look like it's being build in CI so that's alsomething that should e looked into.

### CODE REVIEW

**Comment by SgrrZhf** [comment]
You also need to change these four files, otherwise these samples will report build error on `fvp_base_revc_2xaemv8a` and `fvp_baser_aemv8r` platforms.

https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/dhcpv4_client/boards/fvp_base_revc_2xaemv8a.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/dhcpv4_client/boards/fvp_baser_aemv8r.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/zperf/boards/fvp_base_revc_2xaemv8a.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/zperf/boards/fvp_baser_aemv8r.overlay

**Comment by SgrrZhf** [comment]
The other part looks good to me.

**Comment by fabiobaltieri** [comment]
> You also need to change these four files, otherwise these samples will report build error on `fvp_base_revc_2xaemv8a` and `fvp_baser_aemv8r` platforms.
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/dhcpv4_client/boards/fvp_base_revc_2xaemv8a.overlay https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/dhcpv4_client/boards/fvp_baser_aemv8r.overlay https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/zperf/boards/fvp_base_revc_2xaemv8a.overlay https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/zperf/boards/fvp_baser_aemv8r.overlay

Thanks @SgrrZhf, I think those are fine since the labels are the same but I did miss out a `phy-handle` in `./dts/arm64/fvp/fvp-aemv8r.dtsi`. Added that up now all of these builds:

```
west build -p -b fvp_base_revc_2xaemv8a samples/net/zperf
west build -p -b fvp_baser_aemv8r samples/net/zperf
west build -p -b fvp_base_revc_2xaemv8a samples/net/dhcpv4_client
west build -p -b fvp_baser_aemv8r samples/net/dhcpv4_client
```

I could still move `phy` under `mdio` in those files if you want me to, end result is the same, I have no preference.

**Comment by SgrrZhf** [comment]
> I could still move phy under mdio in those files if you want me to, end result is the same, I have no preference.

Thanks, you don't need to do that, it's okay for me.

**Review by SgrrZhf — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
@tbursztyka any chance you could check this out? Got a couple PRs blocked by it. Thanks!

File: boards/arm64/fvp_base_revc_2xaemv8a/fvp_base_revc_2xaemv8a.dts [hunk 1]
```diff
@@ -132,22 +132,27 @@
...
+		ethernet@1a000000 {
 			reg = <0x1a000000 0x1000>;
-			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
-			status = "disabled";
 
-			phy: phy {
-				compatible = "ethernet-phy";
+			eth: ethernet {
+				compatible = "smsc,lan91c111";
+				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
```
Comment (gmarull) [inline boards/arm64/fvp_base_revc_2xaemv8a/fvp_base_revc_2xaemv8a.dts:140]: doesn't look this binding is aligned with Linux, any reason why? https://elixir.bootlin.com/linux/v6.4.8/source/Documentation/devicetree/bindings/net/smsc,lan91c111.yaml#L4

Comment (fabiobaltieri) [reply boards/arm64/fvp_base_revc_2xaemv8a/fvp_base_revc_2xaemv8a.dts:140]: Does not look like the Linux one support specifying an mdio bus and phy. I could remove the parent node and specify the address on both nodes if you think that's better, that'll bring it closer but it's a bit silly to repeat it, this works well with how the Zephyr driver is written.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**